codebuild/bin/install_default_dependencies.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+set -ex
+source codebuild/bin/s2n_setup_env.sh
+
+ # Install latest version of clang, clang++, and llvm-symbolizer. Needed for fuzzing.
+if [[ "$TESTS" == "fuzz" || "$TESTS" == "ALL" || "$LATEST_CLAG" == "true" ]] && [[ ! -d "$LATEST_CLANG_INSTALL_DIR" ]]; then
+    mkdir -p "$LATEST_CLANG_INSTALL_DIR";
+    codebuild/bin/install_clang.sh "$(mktemp -d)" "$LATEST_CLANG_INSTALL_DIR" "$OS_NAME" > /dev/null ;
+fi
+
+# Download and Install LibFuzzer with latest clang
+if [[ "$TESTS" == "fuzz" || "$TESTS" == "ALL" ]] && [[ ! -d "$LIBFUZZER_INSTALL_DIR" ]]; then
+    mkdir -p "$LIBFUZZER_INSTALL_DIR";
+    PATH=$LATEST_CLANG_INSTALL_DIR/bin:$PATH codebuild/bin/install_libFuzzer.sh "$(mktemp -d)" "$LIBFUZZER_INSTALL_DIR" "$OS_NAME" > /dev/null ;
+fi
+
+# Download and Install Openssl 1.1.1
+if [[ ("$S2N_LIBCRYPTO" == "openssl-1.1.1") || ("$TESTS" == "integration"  || "$TESTS" == "ALL" ) ]] && [[ ! -d "$OPENSSL_1_1_1_INSTALL_DIR" ]]; then
```

Comment:
Check for if a directory exists when the directory is created whether or not the install succeeds means that even a failed install will create the directory. Then, if the install is attempted again this check will not run it, assuming that it is successfully installed even though the folder is empty.